addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -67,6 +68,15 @@ export class ProductScreen extends Component {
             if (this.currentOrder?.state !== "draft") {
                 this.pos.add_new_order();
             }
+
+            if (!this.pos.data.network.loading) {
```

Comment:
Can you explain why its needed ?

Reply:
@davidmonnom 
Without this, the code runs infinitely whenever we make a call, such as `_getProductByBarcode`, `create_product_variant_from_pos`, or `when closing a session from the product screen`. 
ultimately break the entire POS.

Reply:
Mmh I see can you try something like this:

```js
        useEffect(
            () => {
                this.state.quantityByProductTmplId = this.currentOrder?.lines?.reduce((acc, ol) => {
                    acc[ol.product_id.product_tmpl_id.id]
                        ? (acc[ol.product_id.product_tmpl_id.id] += ol.qty)
                        : (acc[ol.product_id.product_tmpl_id.id] = ol.qty);
                    return acc;
                }, {});
            },
            () => [this.currentOrder.totalQuantity]
        );

// pos_order.js
    get totalQuantity() {
        return this.lines.reduce((sum, line) => sum + line.get_quantity(), 0);
    }
```

Because even with the check onWillRender is currently looping a lot.

Reply:
@davidmonnom ,
Changes are done.
It solves the looping problem.
Could you please check?